apis/datadoghq/v2alpha1/datadogagent_default.go

```diff
@@ -142,6 +144,10 @@ func defaultGlobalConfig(ddaSpec *DatadogAgentSpec) {
 		ddaSpec.Global.LogLevel = apiutils.NewStringPointer(defaultLogLevel)
 	}
 
+	if ddaSpec.Global.OriginDetectionUnified == nil {
+		ddaSpec.Global.OriginDetectionUnified = apiutils.NewBoolPointer(defaultOriginDetectionUnified)
+	}
```

Comment:
by adding a default value here. it means that the DatadogAgent resource will be patch to add the info into the spec. 

For this specific option, I'm not sure it is the best solution since the default value is set to "false" . so it means that if later we set the default to `true`. it will not be taken into consideration. 

So usually with "default" an option when it is not an option on which we want to change the default value.

instead in the DSD feature logic you should check is the option is nil then it is equal to the default value.

Reply:
Done, although it might be possible to optimize the code.